### ISSUE TITLE
delete 'browserify' task in watch task?

### ISSUE THREAD

**Issue by prugel** [jakemmarsh/angularjs-gulp-browserify-boilerplate#12]
I think 'browserify' task is duplicate in 'watch' task.

I suggest that 
```
'use strict';

var config        = require('../config');
var gulp          = require('gulp');

gulp.task('watch', ['browserSync', 'server'], function() {
  gulp.watch(config.scripts.src, ['lint']);  // remove 'browserify' task
  gulp.watch(config.styles.src,  ['styles']);
  gulp.watch(config.images.src,  ['images', 'reload']);
  gulp.watch(config.views.watch, ['views']);

  gulp.watch(config.scripts.src, function (event) {
       // auto edit _index.js files 
       if (event.type === 'added' || event.type === 'deleted') {
           autoEditAngular_index( event.path+'/../..');   //  or   'app/js'
       }
       
  });
});

function autoEditAngular_index (dir) {
     // each constants, directives... 
     // edit _index.js
}
```

1. some file edit -> lint 
                             -> bundler.on('update', function() { rebundle(); });
2. some file added -> lint 
                                 -> autoEditAugular_index
                                   -> bundler.on('update', function() { rebundle(); });
3. some file deleted -> ... same 2

How about this?

**Comment by jakemmarsh** [comment]
I'm not sure how this is simpler than the current watch setup. To me, it seems to add unnecessary complexity. How is the browserify task "duplicate" in the watch task?

**Comment by rwam** [comment]
It's more than a multiplier and I think it's a general problem. The more changes I do the more rebundles will occur. I start the dev task and save a file. My output looks like this:

```
[11:19:06] Starting 'views'...
[11:19:06] Finished 'views' after 8.42 ms
[11:19:06] Starting 'lint'...
[11:19:06] Starting 'browserify'...
[11:19:06] Rebundle...
[11:19:06] Finished 'lint' after 310 ms
[11:19:07] Rebundle...
[BS] Reloading Browsers...
[BS] 0 file changed ()
[11:19:07] Finished 'browserify' after 877 ms
GET / 200 8.492 ms - 465
GET /css/main.css 304 1.621 ms - -
GET /js/main.js 200 5.601 ms - 2223
[BS] Reloading Browsers...
[BS] 0 file changed ()
GET /favicon.ico 200 1.545 ms - 465
```

I do another save:

```
[11:19:26] Starting 'views'...
[11:19:26] Finished 'views' after 12 ms
[11:19:26] Starting 'lint'...
[11:19:26] Starting 'browserify'...
[11:19:26] Rebundle...
[11:19:27] Finished 'lint' after 250 ms
[11:19:27] Rebundle...
[11:19:27] Rebundle...
[BS] Reloading Browsers...
[BS] 0 file changed ()
[11:19:27] Finished 'browserify' after 722 ms
GET / 200 11.812 ms - 465
GET /css/main.css 304 1.330 ms - -
GET /js/main.js 200 4.901 ms - 2223
[BS] Reloading Browsers...
[BS] 0 file changed ()
[BS] Reloading Browsers...
[BS] 0 file changed ()
GET /favicon.ico 200 1.903 ms - 465
```

I make a change:

```
[11:19:56] Starting 'views'...
[11:19:56] Finished 'views' after 6.06 ms
[11:19:56] Starting 'lint'...
[11:19:56] Starting 'browserify'...
[11:19:56] Rebundle...
[11:19:56] Finished 'lint' after 223 ms
[11:19:57] Rebundle...
[11:19:57] Rebundle...
[11:19:57] Rebundle...
[BS] Reloading Browsers...
[BS] 0 file changed ()
[11:19:57] Finished 'browserify' after 807 ms
GET / 200 317.101 ms - 465
GET /css/main.css 304 1.734 ms - -
GET /js/main.js 200 5.413 ms - 2781
[BS] Reloading Browsers...
[BS] 0 file changed ()
[BS] Reloading Browsers...
[BS] 0 file changed ()
[BS] Reloading Browsers...
[BS] 0 file changed ()
GET /favicon.ico 200 1.863 ms - 465
```

and so on…

Furthermore the multiple rebundles crashed the dev task. Actually I stop the dev task after 10 changes and start it again.

hm...

**Comment by rwam** [comment]
At first I remove the browserify task from the watch task like @prugel

**Comment by jakemmarsh** [comment]
Ah, I see what you guys mean now. Should be fixed by eb5fa5906d121e67f920f71059287b25abe8237f
